### ISSUE TITLE
Possible divergence between 1.11 and MNB

### ISSUE THREAD

**Issue by Lichtso** [solana-labs/solana#26150]
#### Problem
During the JIT-vs-Interpreter experiment one bare metal node (dev-server-asia-sg1-2) which was running 1.11 (fc166daa1709dd3dd6653bb57c37037412dd7690 from Jun 13) against MNB stopped making roots. Another bare metal node (dev-server-asia-hk2) however, which ran the same configuration, did not fail.

At the time MNB was running on 1.10 for the most part, with a small minority (estimated around 10%-20%) still on 1.9.

The last block was [138053340](https://explorer.solana.com/block/138053340) produced on Jun 18, 2022 at 19:57:21 UTC, which is well inside its epoch (Jun 16, 2022 at 21:44:21 UTC to Jun 20, 2022 at 04:39:11 UTC). After that the node restarted continuously and went out of memory.

#### Proposed Solution
Investigate, debug, fix

**Comment by jeffwashington** [comment]
Well, in this case it turns out that both machines in this case were oom restarting. One of them restarts about twice a day for the past week, catches up, then runs for a while making roots until it ooms again. The other one (sg1-2) is currently stuck unable to start before quickly oom. Good news in that at least one issue here is less intermittent than it looked. Should make it easier to track down.

**Comment by jeffwashington** [comment]
still looking. will be afk for a few hours. Looks like shrink stops shrinking. Perhaps there is a big backlog. but flush, clean, and hash calc still keep periodically working, but shrink stops working. avg storage size begins rapidly increasing. memory starts being consumed rapidly, and we oom.

**Comment by jeffwashington** [comment]
still looking. I am narrowing in.
I found and fixed a separate problem that I have seen which had similar enough behavior. That problem was in pre-release cli options (non-default). So, I have cleared up the picture and know better the issues are unrelated. I have restarted more validators with the fixed pre-release code running so there is a better chance of the current likely master bug returning again with more validators attempting to reproduce it.
